vowpalwabbit/learner.cc

```diff
@@ -196,13 +208,20 @@ class multi_example_handler
     if (!ec_seq.empty()) { _context.template process<multi_ex, learn_multi_ex>(ec_seq); }
   }
 
-  void on_example(example* ec)
+  void on_example(std::vector<example*>* ev)
   {
-    if (try_complete_multi_ex(ec))
+    for (example* ec : *ev)
     {
-      _context.template process<multi_ex, learn_multi_ex>(ec_seq);
-      ec_seq.clear();
+      work_on_example(_context.get_master(), ec);
+      
+      if (try_complete_multi_ex(ec))
```

Comment:
you'll need to rewrite the `try_complete_multi_ex()` function. Its taking an example, and if its part of a multi-example then its compiling them into another vector that will get sent down the learning stack. Thats inefficient, we should just send the vector we already have.

Basically, what we should be doing is:
```
if (ev[0]->indices.size() > 1)
  complete_multi_ex(ev)
  process<learn_multi_ex>()
else if (ev[0]->end_pass)
  process<end_pass>()
else if (is_save_cmd(ev[0]))
  process<save>()
else
  complete_multi_ex(ev)
  process<learn_multi_ex>
```

complete_multi_ex would need a bit of work in this case

Reply:
Ive made a slightly different change for this, please let me know if that works. (Although there's a object_pool assertion which needs to be solved)